FetchRequests/Sources/Controller/FetchedResultsController.swift

```diff
@@ -364,11 +388,35 @@ private extension FetchedResultsController {
             return holder.value
         }
 
-        guard let index = fetchedObjects.firstIndex(where: { $0.id == objectID }) else {
+        if !Thread.isMainThread {
+            DispatchQueue.main.async { [weak self] in
```

Comment:
`performOnMainThread` would need to support throwing in an awkward way if I wanted to reuse it here.